### PR TITLE
Rename lsif-server to precise-code-intel

### PR DESCRIPTION
**Pull request by efritz**
Get rid of the term LSIF where it's just a minor implementation detail.

### CODE REVIEW

File: babel.config.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ module.exports = api => {
     ],
     plugins: [
       'babel-plugin-lodash',
-      // Required to support typeoerm decorators in ./cmd/lsif-server
+      // Required to support typeoerm decorators in ./cmd/precise-code-intel
```
Comment (creachadair) [inline babel.config.js]:
```suggestion
      // Required to support typeorm decorators in ./cmd/precise-code-intel
```

…I think?

File: cmd/precise-code-intel/main.go [hunk 2]
```diff
@@ -14,9 +14,9 @@ import (
 )
 
 var (
-	apis           = env.Get("LSIF_NUM_APIS", "1", "the number of API instances to run (defaults to one)")
-	bundleManagers = env.Get("LSIF_NUM_BUNDLE_MANAGERS", "1", "the number of bundle manager instances to run (defaults to one)")
-	workers        = env.Get("LSIF_NUM_WORKERS", "1", "the number of worker instances to run (defaults to one)")
+	apis           = env.Get("NUM_APIS", "1", "the number of API instances to run (defaults to one)")
```
Comment (creachadair) [inline cmd/precise-code-intel/main.go]: Will these unqualified variable names cause any problems when the code intelligence system is deployed with the rest of Sourcegraph? I guess the name "bundle manager" is probably unique, but "WORKERS" seems pretty generic.

A quick grep only showed `CAMPAIGNS_MAX_WORKERS` so maybe we're safe, but this makes me think we ought to keep a prefix—perhaps `PCI_*`?

Comment (creachadair) [reply cmd/precise-code-intel/main.go]: An advance defense for concern about the abbreviation: `LSIF_*` was already an initialism, so `PCI_*` should also work (as long as we update our docs accordingly).

File: cmd/precise-code-intel/src/api-server/backend/database.ts [hunk 3]
```diff
@@ -167,7 +167,7 @@ export class Database {
     //
 
     private async request<T>(method: string, searchParams: URLSearchParams, ctx: TracingContext): Promise<T> {
-        const url = new URL(`/dbs/${this.dumpId}/${method}`, settings.LSIF_BUNDLE_MANAGER_URL)
+        const url = new URL(`/dbs/${this.dumpId}/${method}`, settings.PRECISE_CODE_INTEL_BUNDLE_MANAGER_URL)
```
Comment (creachadair) [inline cmd/precise-code-intel/src/api-server/backend/database.ts]:
```suggestion
        const url = new URL(`/dbs/${this.dumpId}/${method}`, settings.PCI_BUNDLE_MANAGER_URL)
```

maybe?

File: cmd/precise-code-intel/main.go (hunk 2 above)
Comment (efritz) [reply cmd/precise-code-intel/main.go]: Good point. I forgot similarly-named variables in other parts of the app.

File: cmd/precise-code-intel/src/api-server/backend/database.ts (hunk 3 above)
Comment (efritz) [reply cmd/precise-code-intel/src/api-server/backend/database.ts]: Updating.

**Comment by efritz** [comment]
@unknwon No, just waiting for k8s to become sane so we can move forward.

**Comment by slimsag** [comment]
please please please, I beg of you, do not introduce a new acronym `pci`, see https://github.com/sourcegraph/deploy-sourcegraph/pull/574#discussion_r401174379

Spelling out "precise code intel" everywhere is 100x better in my book.

**Pull request merged**